client/app/pages/dashboards/dashboard.html

```diff
@@ -110,8 +110,9 @@ <h2>
       <span class="hidden-xs hidden-sm">Widgets are individual query visualizations or text boxes you can place on your dashboard in various arrangements.</span>
     </h2>
     <div>
-      <a class="btn btn-default" ng-click="$ctrl.addWidget('textbox')">Add Textbox</a>
-      <a class="btn btn-primary m-l-10" ng-click="$ctrl.addWidget('widget')">Add Widget</a>
+      <a class="btn btn-default" ng-click="$ctrl.addTextBox()">Add Textbox</a>
+      <a class="btn btn-primary m-l-10" ng-click="$ctrl.openAddWidgetDialog()">Add Widget</a>
+      <add-widget-dialog ng-if="$ctrl.addWidgetDialogOpened" dashboard="$ctrl.dashboard" on-close="$ctrl.closeAddWidgetDialog" on-added="$ctrl.onWidgetAdded" />
```

Comment:
It's weird that similar actions are triggered by handlers with so different names (`addTextBox` vs. `openAddWidgetDialog`), but (as I understand) it's because "Add Textbox" modal is still Bootstrap modal - am I right?